Plugins/Public/playercntl_plugin/PimpShip.cpp

```diff
@@ -180,6 +179,23 @@ namespace PimpShip
 			}
 		}
 
+		mapInfo[iClientID].bInPimpDealer = true;
+
+		// Build the equipment list.
+		int iSlotID = 1;
+
+		list<EquipDesc> &eqLst = Players[iClientID].equipDescList.equip;
+		for (list<EquipDesc>::iterator eq = eqLst.begin(); eq != eqLst.end(); eq++)
```

Comment:
Before we enter this loop we should probably `mapInfo[iClientID].mapCurrEquip.clear();`?

Reply:
Oh, yes. And rebuild equip list after purchasing new ship.